packages/data-stores/src/plans/test/selectors.ts

```diff
@@ -0,0 +1,326 @@
+		// Plan slug is undefined
+		expect(
+			Selectors.getPlanByPeriodAgnosticSlug( mockState, undefined, TEST_LOCALE_1 )
+		).toBeUndefined();
+
+		// A plan that doesn't exist in the mock APIs
+		expect(
+			Selectors.getPlanByPeriodAgnosticSlug( mockState, 'ecommerce', TEST_LOCALE_1 )
+		).toBeUndefined();
+
+		// A plan that exists in the store, for locale 1
+		expect(
+			Selectors.getPlanByPeriodAgnosticSlug(
+				mockState,
+				MockData.STORE_PLAN_PREMIUM.periodAgnosticSlug,
+				TEST_LOCALE_1
+			)
+		).toEqual( MockData.STORE_PLAN_PREMIUM );
+
+		// A plan that exists in the store, for locale 2
+		expect(
+			Selectors.getPlanByPeriodAgnosticSlug(
+				mockState,
+				MockData.STORE_PLAN_FREE.periodAgnosticSlug,
+				TEST_LOCALE_2
+			)
+		).toEqual( MockData.STORE_PLAN_FREE );
+
+		// Plan exists in the store, but not for locale 2
+		expect(
+			Selectors.getPlanByPeriodAgnosticSlug(
+				mockState,
+				MockData.STORE_PLAN_PREMIUM.periodAgnosticSlug,
+				TEST_LOCALE_2
+			)
+		).toBeUndefined();
+	} );
+
+	it( 'getDefaultPaidPlan', () => {
+		// The store has the default paid plan for the selected locale
+		expect( Selectors.getDefaultPaidPlan( mockState, TEST_LOCALE_1 ) ).toEqual(
+			MockData.STORE_PLAN_PREMIUM
+		);
+
+		// The store doesn't have the default paid plan for the selected locale
+		expect( Selectors.getDefaultPaidPlan( mockState, TEST_LOCALE_2 ) ).toBeUndefined();
+	} );
+
+	it( 'getDefaultFreePlan', () => {
+		expect( Selectors.getDefaultFreePlan( mockState, TEST_LOCALE_1 ) ).toEqual(
+			MockData.STORE_PLAN_FREE
+		);
+		expect( Selectors.getDefaultFreePlan( mockState, TEST_LOCALE_2 ) ).toEqual(
+			MockData.STORE_PLAN_FREE
+		);
+	} );
+
+	it( 'getSupportedPlans', () => {
+		expect( Selectors.getSupportedPlans( mockState, TEST_LOCALE_1 ) ).toEqual(
+			mockPlans[ TEST_LOCALE_1 ]
+		);
+		expect( Selectors.getSupportedPlans( mockState, TEST_LOCALE_2 ) ).toEqual(
+			mockPlans[ TEST_LOCALE_2 ]
+		);
+		expect( Selectors.getSupportedPlans( mockState, 'non-existing' ) ).toEqual( [] );
+	} );
+
+	it( 'getPlansProducts', () => {
+		expect( Selectors.getPlansProducts( mockState ) ).toEqual( mockPlanProducts );
+	} );
+
+	it( 'getPrices', () => {
+		const expectedPrices = {
+			[ MockData.STORE_PRODUCT_FREE.storeSlug ]: MockData.STORE_PRODUCT_FREE.price,
+			[ MockData.STORE_PRODUCT_PREMIUM_ANNUALLY.storeSlug ]:
+				MockData.STORE_PRODUCT_PREMIUM_ANNUALLY.price,
+		};
+		expect( Selectors.getPrices( mockState, TEST_LOCALE_1 ) ).toEqual( expectedPrices );
+		expect( Selectors.getPrices( mockState, TEST_LOCALE_2 ) ).toEqual( expectedPrices );
+
+		// Make sure function is correctly flagged as deprecated
+		const expectedCallArguments = [
+			'getPrices',
+			{
+				alternative: 'getPlanProduct().price',
+			},
+		];
+		expect( deprecate ).toHaveBeenCalledTimes( 2 );
+		expect( deprecate ).toHaveBeenNthCalledWith( 1, ...expectedCallArguments );
+		expect( deprecate ).toHaveBeenNthCalledWith( 2, ...expectedCallArguments );
```

Comment:
What value do these test cases add? In other words: Are we testing our selectors or `@wordpress/deprecated `?

Reply:
We're testing that this specific selector is correctly flagged as deprecated by checking that the function responsible for the warning is being called correctly. We're not testing `@wordpress/deprecated` at all, since it's mocked.